File: templates/template.py
"""
Problem: {PROBLEM_NAME}
Difficulty: {DIFFICULTY}
Date: {DATE}
Language: Python
Approach: {APPROACH}
Time: O({TIME_COMPLEXITY})
Space: O({SPACE_COMPLEXITY})
"""

def {FUNCTION_NAME}({PARAMETERS}):
    """
    {PROBLEM_DESCRIPTION}
    
    Args:
        {ARGS_DESCRIPTION}
    
    Returns:
        {RETURN_DESCRIPTION}
    """
    # TODO: Implement solution
    pass

if __name__ == "__main__":
    # Test cases
    test_cases = [
        # ({INPUT}, {EXPECTED_OUTPUT}),
    ]
    
    for i, (input_data, expected) in enumerate(test_cases):
        result = {FUNCTION_NAME}(input_data)
        print(f"Test {i+1}: {'PASS' if result == expected else 'FAIL'}")
        print(f"  Input: {input_data}")
        print(f"  Expected: {expected}")
        print(f"  Got: {result}")

File: templates/template.cpp
/**
 * Problem: {PROBLEM_NAME}
 * Difficulty: {DIFFICULTY}
 * Date: {DATE}
 * Language: C++
 * Approach: {APPROACH}
 * Time: O({TIME_COMPLEXITY})
 * Space: O({SPACE_COMPLEXITY})
 */

#include <iostream>
#include <vector>
#include <cassert>

class Solution {
public:
    /**
     * {PROBLEM_DESCRIPTION}
     * 
     * @param {PARAM_TYPE} {PARAMETER} {PARAM_DESCRIPTION}
     * @return {RETURN_TYPE} {RETURN_DESCRIPTION}
     */
    {RETURN_TYPE} {FUNCTION_NAME}({PARAM_TYPE} {PARAMETER}) {
        // TODO: Implement solution
    }
};

int main() {
    Solution solution;
    
    // Test cases
    struct TestCase {
        {PARAM_TYPE} input;
        {RETURN_TYPE} expected;
    };
    
    std::vector<TestCase> testCases = {
        // {{INPUT}, {EXPECTED_OUTPUT}},
    };
    
    for (size_t i = 0; i < testCases.size(); ++i) {
        auto result = solution.{FUNCTION_NAME}(testCases[i].input);
        bool passed = (result == testCases[i].expected);
        
        std::cout << "Test " << (i + 1) << ": " << (passed ? "PASS" : "FAIL") << std::endl;
        std::cout << "  Input: " << testCases[i].input << std::endl;
        std::cout << "  Expected: " << testCases[i].expected << std::endl;
        std::cout << "  Got: " << result << std::endl;
    }
    
    return 0;
}
